### PR TITLE
SE-10138: Cross-lingual handling of salutations

### PR DESCRIPTION
**Pull request by marvinwoop**
Since it is possible in S2 to send e-mails to several addressees in different languages, they must be addressable in their respective language.

### CODE REVIEW

File: src/main/java/sirius/biz/model/PersonData.java [hunk 1]
```diff
@@ -212,6 +225,16 @@ public String getTranslatedSalutation() {
         return codeLists.getTranslatedValue(CODE_LIST_SALUTATIONS, salutation);
     }
 
+    /**
+     * Returns the value (by langCode specified translated name) of the salutation.
+     *
+     * @param langCode the language code to translate to
+     * @return the value for <tt>salutation</tt> from the <tt>salutations</tt> code list
+     */
+    public String getUserSpecificTranslatedSalutation(String langCode) {
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/model/PersonData.java:234]:
```suggestion
    public String getTranslatedSalutation(String langCode) {
```

File: src/main/java/sirius/biz/codelists/CodeLists.java [hunk 2]
```diff
@@ -438,7 +438,21 @@ public String getValue(@Nonnull String codeListName, @Nullable String code, @Nul
...
+     *
+     * @param codeListName the code list to search in
+     * @param code         the code to lookup
+     * @param langCode     the language code to translate into
+     * @return the translated value associated with the code or the code itself if no value exists
+     */
+    public String getTranslatedValue(@Nonnull String codeListName, @Nullable String code, String langCode) {
+        String value = getValue(codeListName, code, langCode);
         if (value != null && value.startsWith("$")) {
             return Value.of(value).translate().getString();
```
Comment (mkeckmkeck) [inline src/main/java/sirius/biz/codelists/CodeLists.java:457]: Don't we need the langcode for Value.of(value).translate() too?

**Review by mkeckmkeck — CHANGES_REQUESTED** [review]
Make naming more consistent

**Review by sabieber — APPROVED** [review]

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/model/PersonData.java (hunk 1 above)
Comment (marvinwoop) [reply src/main/java/sirius/biz/model/PersonData.java]: Renamed method.

File: src/main/java/sirius/biz/codelists/CodeLists.java (hunk 2 above)
Comment (marvinwoop) [reply src/main/java/sirius/biz/codelists/CodeLists.java]: Yes we do actually need that code in that method. Added it.

**Review by idlira — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Pull request merged**